code/modules/mob/living/carbon/xenomorph/abilities/burrower/burrower_powers.dm

```diff
@@ -209,3 +271,270 @@
+			playsound(get_turf(C), "alien_bite", 50, TRUE)
+		for(var/obj/structure/S in target_turf)
+			if(istype(S, /obj/structure/window/framed))
+				var/obj/structure/window/framed/W = S
+				if(!W.unslashable)
+					W.shatter_window(TRUE)
+					playsound(target_turf, "windowshatter", 50, TRUE)
+		sleep(chain_separation_delay)
+
+
+/datum/action/xeno_action/activable/sunken_tail/use_ability(atom/A)
+	var/mob/living/carbon/Xenomorph/X = owner
+	if(!istype(X))
+		return
+
+	if(!action_cooldown_check())
+		return
+
+	if(!A || A.layer >= FLY_LAYER || !X.check_state() || X.action_busy)
+		return
+
+	if(X.burrow)
+		to_chat(X, SPAN_XENOWARNING("You can't do this while burrowed!"))
+		return
+
+	var/distance = max_distance
+	var/damage = base_damage
+	if(X.fortify)
+		distance += reinforced_range_bonus
+		damage += reinforced_damage_bonus
+
+	if(get_dist(A, X) > distance)
+		to_chat(X, SPAN_XENOWARNING("[A] is too far away!"))
+		return
+
+	if(!check_clear_path_to_target(X, A, FALSE))
+		to_chat(X, SPAN_XENOWARNING("Something is blocking our path to [A]!"))
+		return
+
+	if(!check_plasma_owner())
+		return
+
+	var/turf/target = locate(A.x, A.y, A.z)
+	var/list/telegraph_atom_list = list()
+	telegraph_atom_list += new /obj/effect/xenomorph/xeno_telegraph/red(target, windup_delay)
+
+	if(!do_after(X, windup_delay, INTERRUPT_ALL | BEHAVIOR_IMMOBILE, BUSY_ICON_HOSTILE))
+		apply_cooldown_override(5 SECONDS)
+		for(var/obj/effect/tele in telegraph_atom_list)
+			qdel(tele)
+		return
+	use_plasma_owner()
+	playsound(X.loc, 'sound/effects/burrower_attack1.ogg', 40)
+	X.visible_message(SPAN_XENOWARNING("The [X] stabs its tail in the ground toward [A]!"), SPAN_XENOWARNING("You stab your tail into the ground toward [A]!"))
+	telegraph_atom_list += new /obj/effect/xenomorph/ground_spike(target)
+	for (var/mob/living/carbon/C in target)
+		if (C.stat == DEAD || HAS_TRAIT(C, TRAIT_NESTED))
+			continue
+
+		if(X.can_not_harm(C))
+			continue
+		C.apply_armoured_damage(damage, ARMOR_MELEE, BRUTE)
+		to_chat(C, SPAN_WARNING("You are stabbed with a tail from below!"))
+		playsound(get_turf(C), "alien_bite", 50, TRUE)
+	for(var/obj/structure/S in target)
+		if(istype(S, /obj/structure/window/framed))
+			var/obj/structure/window/framed/W = S
+			if(!W.unslashable)
+				W.shatter_window(TRUE)
+				playsound(target, "windowshatter", 50, TRUE)
+
+	apply_cooldown()
+	..()
+	return
+
+/datum/action/xeno_action/activable/ensconce/use_ability()
+	var/mob/living/carbon/Xenomorph/xeno = owner
+	if(!istype(xeno))
+		return
+
+	if(!action_cooldown_check())
+		return
+
+	if(!xeno.check_state()|| xeno.action_busy)
+		return
+
+	if(xeno.burrow)
+		to_chat(xeno, SPAN_XENOWARNING("You can't do this while burrowed!"))
+		return
+
+	var/turf/T = get_turf(xeno)
+	var/obj/effect/alien/weeds/weeds = locate() in T
+	if(!weeds || !xeno.ally_of_hivenumber(weeds.hivenumber))
+		to_chat(xeno, SPAN_XENOWARNING("You need to do this on weeds!"))
+		return
+
+	if(!check_plasma_owner())
```

Comment:
isn't plasma cost zero?

if you want to support changing it in the future, might want to also call check_and_use_plasma_owner

Reply:
I want to change plasma after the do_after, just incase someone wants to put a plasma cost on it via ability.
Done via line 460